### PR TITLE
Add custom error check to hierarchical clustering for too many rows N >= 65536

### PR DESCRIPTION
**Pull request by koenderks**
Due to the lines:

`if (is.na(n) || n > 65536L)`
`        stop("size cannot be NA nor exceed 65536")`

in the `hclust` function from the `stats` package.

### CODE REVIEW

**Comment by koenderks** [comment]
Also made some changes to the help files for clustering

File: Resources/Help/analyses/mlclusteringdensitybased.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Density-based clustering is a soft clustering method where clusters are construc
...
-- T-sne cluster plot: Generates a T-sne plot of the clustering output. T-sne plots are used for visualizing high-dimensional data in a low-dimensional space of two dimensions aiming to illustrate the relative distances between data observations. The T-sne two-dimensional space makes the axes uninterpretable. A T-sne plot seeks to give an impression of the relative distances between observations and clusters
+- t-SNE cluster plot: Generates a t-SNE plot of the clustering output. t-SNE plots are used for visualizing high-dimensional data in a low-dimensional space of two dimensions aiming to illustrate the relative distances between data observations. The T-sne two-dimensional space makes the axes uninterpretable. A t-SNE plot seeks to give an impression of the relative distances between observations and clusters. To recreate the same t-SNE plot across several clustering analyses you can set their seed to the same value, as the t-SNE algorithm uses random starting values.
```
Comment (AlexanderLyNL) [inline Resources/Help/analyses/mlclusteringdensitybased.md]: The T-sne two-dimensional space makes the axes uninterpretable. 

The T-SNE ....

Comment (AlexanderLyNL) [reply Resources/Help/analyses/mlclusteringdensitybased.md]: Same in the other two files

File: JASP-Engine/JASP/R/commonMachineLearningClustering.R [hunk 2]
```diff
@@ -38,11 +48,15 @@
...
-  if(length(predictors[predictors != ""]) > 0L)
-    .hasErrors(dataset, type = c('infinity', 'observations'), custom=checkClusters, all.target = predictors,
-               observations.amount = "< 2", exitAnalysisIfErrors = TRUE)
+  checkDataHierarchicalClustering <- function() {
+    if(type != "hierarchical")
+      return()
 
-  return()
+    if(nrow(dataset) >= 65536L)
+      return("R package error: The hclust clustering algorithm from the stats R package cannot handle data that has 65536 or more rows. Please try another algorithm, as we are working towards a solution for this problem.")
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/commonMachineLearningClustering.R]: We are working on a solution for this problem. Please try another algorithm in the meantime.

File: Resources/Help/analyses/mlclusteringdensitybased.md (hunk 1 above)
Comment (koenderks) [reply Resources/Help/analyses/mlclusteringdensitybased.md]: Thanks for pointing out, fixed it now.

**Pull request merged**